src/Dataset.ts

```diff
@@ -577,6 +577,101 @@ export default class Dataset {
     }
     return this._allPrefixes;
   }
+
+  private async apiNDEDatasetRegister(method: "validate" | "submit", next?: (shaclReport: n3.Store) => void) {
+    const info = await this.getInfo();
+    if (info.accessLevel !== "public") {
+      throw getErr(
+        `Only datasets with accesslevel 'public' can be submitted your dataset has accesslevel '${info.accessLevel}'.`
+      );
+    }
+    const { consoleUrl } = await this._app.getInfo();
+    const apiUrl =
+      "https://datasetregister.netwerkdigitaalerfgoed.nl/api/datasets" + (method === "validate" ? "/validate" : "");
```

Comment:
in general, you're constructing a lot of URL and URL paths using string concatenation. Fine if this is done, but we should make sure this does not regress (now or in the future). when in doubt, we should use the URL and path libs instead

Reply:
noted, thnx, maybe we should consider moving away from using strings in TriplyDB-JS, e.g. the `consoleUrl` should return a `URL` not a string.